app/modules/configuration/resources.py

```diff
@@ -39,6 +39,32 @@ def get(self, target, path):
             path,
             target=target,
         )
+        from app.modules.ia_config_reader import IaConfig
+
+        ia_config_reader = IaConfig()
+        species = ia_config_reader.get_configured_species()
+        if not isinstance(
+            data['response']['configuration']['site.species']['suggestedValues'], list
```

Comment:
I had a quick look at this and this API uses a "path" so this piece of code looks like it's only for `if path == 'site.species'`.  This is what edm returns:

```json
{
  "success": true,
  "response": {
    "schema": {
      "displayOrder": 0.5
    },
    "multiple": true,
    "readOnly": false,
    "isPrivate": false,
    "required": true,
    "suggestedValues": [
      { 
        "commonNames": [
          "whale shark",
          "tiburón ballena",
          "dámero",
          "ジンベイザメ"
        ],
        "scientificName": "Rhincodon typus",
        "itisTsn": 159857
      },
      {
        "commonNames": [
          "common bottlenose dolphin"
        ],
        "scientificName": "Tursiops truncatus",
        "itisTsn": 180426
      },
      {
        "commonNames": [
          "sperm whale"
        ],
        "scientificName": "Physeter macrocephalus",
        "itisTsn": 180488
      },
      {
        "commonNames": [
          "humpback whale",
          "rorqual à bosse"
        ],
        "scientificName": "Megaptera novaeangliae",
        "itisTsn": 180530
      }
    ],
    "displayType": "taxonomy",
    "settable": true,
    "descriptionId": "CONFIGURATION_SITE_SPECIES_DESCRIPTION",
    "translationId": "CONFIGURATION_SITE_SPECIES",
    "labelId": "CONFIGURATION_SITE_SPECIES_LABEL",
    "name": "configuration_site_species",
    "parentConfigurationId": "site",
    "configurationId": "site.species",
    "fieldType": "taxonomy"
  },
  "transactionId": "3be5b87f-8163-4828-85db-28c6528db1d1"
}
```

So this line should be `data['response']['suggestedValues']`

We can do this instead so the if statement doesn't span so many lines:

```python
if not isinstance(data['response']['suggestedValues'], list):
    data['response']['suggestedValues'] = []
suggested_values = data['response']['suggestedValues']
suggested_species = [
    x.get('scientificName') for x in suggested_values
]
for sn in species:  # only adds a species that is not already in suggestedValues
    if sn not in suggested_species:
        suggested_values.insert(
            0, {'scientificName': sn, 'commonNames': [sn]}
        )
```

Reply:
ah you are right -- this is being over-applied.  i need something akin to the conditional on line 122:
`if path == '__bundle_setup':` ... it only needs to happen in a couple of specific cases.  i will fix.
(i was just working on this today to try to put in some kind of test; but thanks for your tip here karen!)